src/metadata/TerraformMetadata.js

```diff
@@ -1,8 +1,207 @@
-import { DefaultMetadata } from 'leto-modelizer-plugin-core';
+import Ajv from 'ajv';
+import {
+  ComponentAttributeDefinition,
+  DefaultMetadata,
+  ComponentLinkDefinition,
+} from 'leto-modelizer-plugin-core';
+import TerraformComponentDefinition from 'src/models/TerraformComponentDefinition';
+import Schema from 'src/metadata/ValidationSchema';
 
 /**
  * Class to validate and retrieve components definitions from Terraform metadata.
  */
-class TerraformMetadata extends DefaultMetadata {}
+class TerraformMetadata extends DefaultMetadata {
+  constructor(resources) {
+    super();
+    this.providers = resources.metadata;
+    this.ajv = new Ajv();
+    this.schema = Schema;
+    this.getAttributeDefinition = this.getAttributeDefinition.bind(this);
+  }
+
+  /**
+   * Validate the provided metadata with a schemas.
+   * @return {Boolean} - True if metadata is valid.
+   */
+  validate() {
+    const errors = [];
+    Object.keys(this.providers).forEach((provider) => {
+      const validate = this.ajv.compile(this.schema);
+
+      if (!validate(this.providers[provider])) {
+        errors.push({
+          provider,
+          errors: validate.errors,
+        });
+      }
+    });
+
+    if (errors.length > 0) {
+      throw new Error('Metadata is not valid', { cause: errors });
+    }
+
+    return true;
+  }
+
+  /**
+   * Get all definitions from metadata.
+   * @returns {{components: ComponentDefinition[], links: ComponentLinkDefinition[]}} Definitions.
+   */
+  getDefinitions() {
+    const components = this.getComponentDefinitions();
+    const links = this.getLinkDefinitions(components);
+
+    return {
+      components,
+      links,
+    };
+  }
+
+  /**
+   * Get link definitions from components.
+   * @param {TerraformComponentDefinition[]} components - Component definitions to parse.
```

Comment:
components, component definitions, not clear...

Reply:
done